File: Calorie_Tracker/Calorie_Tracker/Models/ApiModels/Food.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Calorie_Tracker.Models.ApiModels;

public class Food
{
    [JsonPropertyName("_id")]
    public string _id { get; set; }

    [JsonPropertyName("barcode")]
    public string barcode { get; set; }  

    [JsonPropertyName("name")]
    public string name { get; set; }  

    [JsonPropertyName("godkendt")]
    public bool? godkendt { get; set; }  

    [JsonPropertyName("calories")]
    public double? calories { get; set; }  

    [JsonPropertyName("carbonhydrates")]
    public double? carbonhydrates { get; set; }  

    [JsonPropertyName("protein")]
    public double? protein { get; set; }  

    [JsonPropertyName("fat")]
    public double? fat{ get; set;}
}

File: API/models/food.js
// imports mongoose to use to create schema
import mongoose from 'mongoose'

const foodSchema = new mongoose.Schema({
    barcode: {
        type: String,
        required: false
    },
    name: {
        type: String,
        required: false
    },
    godkendt: {
        type: Boolean,
        required: false
    },
    calories: {
        type: Number,
        required: false
    },
    carbonhydrates: {
        type: Number,
        required: false
    },
    protein: {
        type: Number,
        required: false
    },
    fat: {
        type: Number,
        required: false
    }
})

// exports scheme as model to mongoose database and controllers
export const Food = mongoose.model('Food', foodSchema)


